src/query/storages/hive/src/hive_table.rs

```diff
@@ -546,9 +540,8 @@ async fn do_list_files_from_dir(
         }
         match de.mode() {
             ObjectMode::FILE => {
-                // todo, support in opendal#list
                 let filename = path.to_string();
-                let length = get_file_length(operator.clone(), path).await?;
+                let length = de.content_length().unwrap();
```

Comment:
The content_length in `DirEntry` is `best-effort`. Users should get via `metadata` again if the content length is not exist.

Reply:
yes,  get metadata again is more friendly with other storages. and now which storages supports or not supports list with metadata?   for it will effect the performance.

Reply:
> now which storages support or not supports list with metadata

First, It's the internal implementation details that users should not depend on. In the future, OpenDAL MAY provide functions to finish this work (get from de, if not found, get from metadata).

Until now, all object storage services and hdfs will return `content_length` during the list. Other services like `fs` don't.